### PR TITLE
add memory trace and update test

### PR DESCRIPTION
**Pull request by hongyuanyang-uu**
add memory trace and update test

### CODE REVIEW

File: executor/src/lib.rs [hunk 1]
```diff
@@ -690,17 +690,102 @@ impl Process {
...
-                //     addr: *addr,
-                //     clk: cell.clk,
-                //     op: cell.op.clone(),
-                //     value: cell.value.clone(),
-                // };
-                // program.trace.memory.push(trace_cell);
+                debug!("addr:{}, cell:{:?}", addr, cell);
+                if cell.region_prophet.is_one() {
+                    diff_addr_cond =
+                        GoldilocksField::ZERO - GoldilocksField::from_canonical_u64(*addr);
```
Comment (Softcloud88) [inline executor/src/lib.rs:707]: P-addr should be 0-1-addr

File: executor/src/lib.rs [hunk 2]
```diff
@@ -690,17 +690,102 @@ impl Process {
...
+                } else if cell.region_poseidon.is_one() {
+                    diff_addr_cond = GoldilocksField::ZERO
```
Comment (Softcloud88) [inline executor/src/lib.rs:709]: 0-1-(2^32-1)-addr

File: executor/src/lib.rs [hunk 3]
```diff
@@ -690,17 +690,102 @@ impl Process {
...
+                        - GoldilocksField::from_canonical_u64(2 ^ 32)
+                        - GoldilocksField::from_canonical_u64(*addr);
+                } else if cell.region_ecdsa.is_one() {
+                    diff_addr_cond = GoldilocksField::ZERO
```
Comment (Softcloud88) [inline executor/src/lib.rs:713]: 0-1-(2*(2^32-1))-addr

**Review by JiangXb-son — APPROVED** [review]

**Review by spartucus — APPROVED** [review]

File: executor/src/lib.rs [hunk 4]
```diff
@@ -690,17 +690,102 @@ impl Process {
...
+                        clk: GoldilocksField::from_canonical_u64(cell.clk as u64),
+                        is_rw: cell.is_rw,
+                        op: cell.op.clone(),
+                        is_write: cell.is_write,
+                        diff_addr,
+                        diff_addr_inv,
+                        diff_clk,
+                        diff_addr_cond,
+                        filter_looked_for_main: cell.filter_looked_for_main,
+                        rw_addr_unchanged: GoldilocksField::ONE,
```
Comment (Softcloud88) [inline executor/src/lib.rs:778]: rw_addr_unchanged should be all 0 outside of is_rw region.

**Review by peinlcy — APPROVED** [review]

**Pull request merged**